File: minimum-operations-to-make-a-uni-value-grid.java

class Solution {
    public int minOperations(int[][] grid, int x) {
        List<Integer> arr = new ArrayList<>();
        for (int[] row : grid) {
            for (int num : row) {
                arr.add(num);
            }
        }
        
        Collections.sort(arr);
        int median = arr.get(arr.size() / 2);
        int operations = 0;

        for (int num : arr) {
            int diff = Math.abs(num - median);
            if (diff % x != 0) return -1;
            operations += diff / x;
        }
        return operations;
    }
}

File: minimum-operations-to-make-a-uni-value-grid.py
import numpy as np

class Solution:
    def minOperations(self, grid: List[List[int]], x: int) -> int:
        nums = np.array(grid).flatten()
    
        # Check feasibility: all elements must have same remainder modulo x
        if np.any((nums - nums[0]) % x != 0):
            return -1
    
        # The L1 minimizer is the median
        nums.sort()
        median = nums[len(nums) // 2]
    
        # Vectorized operation to find total steps
        return int(np.sum(np.abs(nums - median)) // x)        
